src/base/bittorrent/session.cpp

```diff
@@ -3920,6 +3920,13 @@ void Session::handleTorrentSavingModeChanged(TorrentHandleImpl *const torrent)
 
 void Session::handleTorrentTrackersAdded(TorrentHandleImpl *const torrent, const QVector<TrackerEntry> &newTrackers)
 {
+    // Clear the peer list if it's a private torrent since 
+    // we do not want to keep connecting with peers from old tracker.
+#if (LIBTORRENT_VERSION_NUM >= 10207)
+    if (torrent->isPrivate())
```

Comment:
I'm not sure, but I feel this code should be placed in torrenthandleimpl.cpp (at those related functions) instead.